lib/rest-router.js

```diff
@@ -0,0 +1,37 @@
+module.exports = RestRouter;
+
+var url = require('url');
+var Trie = require('./trie');
+
+function RestRouter(options) {
+  this.trie = new Trie();
+  this.options = options;
+}
+
+//register path and related handler to Trie ds
+RestRouter.prototype.registerPathAndHandlers = function(route, handler) {
+  var path = normalizePath(route.fullPath);
+  var trie = this.trie;
+  trie = trie.add(path, route.verb, handler);
+};
+
+//handle request, match path and if found, invoke handler method
+RestRouter.prototype.handle = function() {
+  var trie = this.trie;
+  return function(req, res, next) {
+    var verb = req.method.toLowerCase() || 'all';
+    var path = normalizePath(req.url, this.options);
+    var methods = trie.find(path);
+    if (methods && verb in methods) {
+      methods[verb](req, res, next);
+    } else next();
```

Comment:
Be consistent with braces - either use them in both "if" and "else" branch, or don't use them in both branches.

I think we tend to use braces unless the whole if + statement fits a single line.

```js
if (...) {
  methods[verb](...);
} else {
  next();
}
```